mm/mm_heap/mm_memalign.c

```diff
@@ -78,7 +79,9 @@ FAR void *mm_memalign(FAR struct mm_heap_s *heap, size_t alignment,
 
   if (alignment <= MM_MIN_CHUNK)
     {
-      return mm_malloc(heap, size);
+      ptr = mm_malloc(heap, size);
+      DEBUGASSERT(ptr == NULL || ((uintptr_t)ptr) % alignment == 0);
+      return ptr;
```

Comment:
```suggestion
      alignedchunk = (size_t)mm_malloc(heap, size);
      DEBUGASSERT(alignedchunk % alignment == 0);
      return (FAR void *)alignedchunk;
```

Reply:
i know (void *)0 is guaranteed to be NULL.
otoh, i'm not sure if (size_t)NULL is guaranteed to be 0.
i feel it's simpler to check NULL.

Reply:
I just want to keep code consistent. We have code at line 112
```
  rawchunk = (size_t)mm_malloc(heap, allocsize);
  if (rawchunk == 0)
    {
      return NULL;
    }
```
so let's keep the same style if possible?

Reply:
do you mean, within this file? or mm/? or the whole repo?

Reply:
The ultimate goad is a whole repo of course :)
But let's focus on `mm_memalign` implementation only here.

Reply:
Oah....I didn't expect that my comments will lead to adding even more local variables :)
I expected changes like
```
uintptr_t rawchunk;
...
rawchunk = (uintptr_t)mm_malloc(heap, allocsize);
if (rawchunk == (uintptr_t)NULL)
  {
    return NULL;
  }
```
and the same here, but not what I see.

Reply:
do you prefer dropping all changes on mm_malloc.c and mm_memalign.c from this PR?
they are not essential parts of this PR. i feel it isn't that constructive to keep discussing about/changing them.

Reply:
At least your previous version was more easy to read. That is my subjective opinion of course.
Agree, let's close the discussion.